### ISSUE TITLE
Cannot convert undefined or null to object

### ISSUE THREAD

**Issue by TheKnarf** [tldraw/tldraw#3155]
### What happened?

I'm trying to load a `.tldr` file at startup. The documentation doesn't seem to document how to do this, the closest I can find is using the `snapshot`-attribute as `<Tldraw snapshot={jsonData} />`. However when doing that I get the following error:

```
Cannot convert undefined or null to object
TypeError: Cannot convert undefined or null to object
    at Function.values (<anonymous>)
    at objectMapValues (webpack-internal:///../../node_modules/.pnpm/@tldraw+utils@2.0.1/node_modules/@tldraw/utils/dist-cjs/lib/object.js:67:17)
    at StoreSchema.migrateStoreSnapshot (webpack-internal:///../../node_modules/.pnpm/@tldraw+store@2.0.1_react@18.2.0/node_modules/@tldraw/store/dist-cjs/lib/StoreSchema.js:135:54)
    at Store.loadSnapshot (webpack-internal:///../../node_modules/.pnpm/@tldraw+store@2.0.1_react@18.2.0/node_modules/@tldraw/store/dist-cjs/lib/Store.js:443:41)
    at createStore (webpack-internal:///../../node_modules/.pnpm/@tldraw+editor@2.0.1_react-dom@18.2.0_react@18.2.0/node_modules/@tldraw/editor/dist-cjs/lib/hooks/useTLStore.js:30:11)
    at eval (webpack-internal:///../../node_modules/.pnpm/@tldraw+editor@2.0.1_react-dom@18.2.0_react@18.2.0/node_modules/@tldraw/editor/dist-cjs/lib/hooks/useTLStore.js:35:66)
    at mountState (webpack-internal:///../../node_modules/.pnpm/react-dom@18.2.0_react@18.2.0/node_modules/react-dom/cjs/react-dom.development.js:16986:20)
    at Object.useState (webpack-internal:///../../node_modules/.pnpm/react-dom@18.2.0_react@18.2.0/node_modules/react-dom/cjs/react-dom.development.js:17699:16)
    at useState (webpack-internal:///../../node_modules/.pnpm/react@18.2.0/node_modules/react/cjs/react.development.js:1623:21)
    at useTLStore (webpack-internal:///../../node_modules/.pnpm/@tldraw+editor@2.0.1_react-dom@18.2.0_react@18.2.0/node_modules/@tldraw/editor/dist-cjs/lib/hooks/useTLStore.js:35:59)
```

Is a `snapshot` something else than a saved `.tldr` file? I'm trying to build a readonly viewer of `.tldr` files for Docusaurus.

### How can we reproduce the bug?

- Setup a Docusaurus project running: `npx create-docusaurus@latest my-website classic`.

- Add dependencies:

```bash
pnpm add tldraw @tldraw/assets json-loader
```

- Create a component called `./src/components/tldr/tldr.jsx`:

```javascript
import BrowserOnly from '@docusaurus/BrowserOnly';
import 'tldraw/tldraw.css'

export const Tldr = ({ children }) => {
  return (
  <>
    <style>{`
	@import url("https://fonts.googleapis.com/css2?family=Inter:wght@500;700&display=swap");

	.tldrawContainer {
		font-family: "Inter";
		background: grey;
		width: 100%;
		height: 500px;
		position: relative;
	}

	.tldrawContainer > div {
		position: absolute;
		inset: 0;
	}
`}</style>
    <div className="tldrawContainer">
      <div>
	<BrowserOnly fallback={<div>Loading...</div>}>
	{() => {
	const { Tldraw } = require('tldraw');
	const { getAssetUrls } = require('@tldraw/assets/selfHosted');

	const assetUrls = getAssetUrls({
			baseUrl: 'https://unpkg.com/@tldraw/assets@2.0.1/',
	});

	console.log(children)

	return <Tldraw
		inferDarkMode
		assetUrls={assetUrls}
		snapshot={children}
		onMount={(editor) => {
			editor.updateInstanceState({ isReadonly: true });
		}}
		/>;
	}}
	</BrowserOnly>
      </div>
    </div>
  </>
  );
}
```

- Setup `json-loader` in the webpack config:

```javascript
module: {
  rules: [
    { test: /\.tldr$/, use: [ 'json-loader' ] },
  ],
},
```

- Create a drawing on tldraw.com and save it locally as `drawing.tldr`.
- Then try and use it in an mdx file:

```mdx
---
sidebar_position: 1
---

import { Tldr } from '../src/components/tldr/tldr.jsx';
import drawing from './drawing.tldr';

# Tldr

<Tldr children={drawing} />
```

### What browsers are you seeing the problem on?

Arc

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by TheKnarf** [comment]
After some help on the tldraw Discord-server I got a some working code:

```javascript
import BrowserOnly from '@docusaurus/BrowserOnly';
import 'tldraw/tldraw.css';
import { useMemo } from 'react';

export const Tldr = ({ children }) => {
  return (
    <>
      <style>{`
  @import url("https://fonts.googleapis.com/css2?family=Inter:wght@500;700&display=swap");

  .tldrawContainer {
    font-family: "Inter";
    background: grey;
    width: 100%;
    height: 500px;
    position: relative;
  }

  .tldrawContainer > div {
    position: absolute;
    inset: 0;
  }
`}</style>
      <div className="tldrawContainer">
        <div>
          <BrowserOnly fallback={<div>Loading...</div>}>
            {() => {
              const { Tldraw, parseTldrawJsonFile, createTLSchema } = require('tldraw');
              const { getAssetUrls } = require('@tldraw/assets/selfHosted');

              const assetUrls = getAssetUrls({
                  baseUrl: 'https://unpkg.com/@tldraw/assets@2.0.1/',
              });

              const result = useMemo(() =>
                parseTldrawJsonFile({ json: JSON.stringify(children), schema: createTLSchema() }),
                [children]
              );

              if (result.err) {
                console.log(result.err);
                return <div>Error loading tldr file</div>;
              }

              console.log(children, result)

              return <Tldraw
                inferDarkMode
                assetUrls={assetUrls}
                store={result.value}
                onMount={(editor) => {
                  editor.updateInstanceState({ isReadonly: true });
                }}
                />;
          }}
          </BrowserOnly>
        </div>
      </div>
    </>
  );
}
```

It would be useful if there was actually some documentation for how to do this. I'm still not sure if my usage of `createTLSchema()` her is the intended usage.
